src/helpers/handleResponse.ts

```diff
@@ -2,11 +2,13 @@ import { STATUS_CODES } from '@/constants/constantValues';
 
 import { convertKeysToCamelCase } from './';
 
-const handleResponse = async <T>(response: Response): Promise<T> => {
+const handleResponse = async <R>(response: Response): Promise<R> => {
   if (response.status === STATUS_CODES.OK) {
     const jsonResponse = await response.json();
 
-    return convertKeysToCamelCase<T>(jsonResponse);
+    return convertKeysToCamelCase<R>(jsonResponse);
+  } else if (response.status === STATUS_CODES.NO_CONTENT) {
+    return null as R;
```

Comment:
If you're retrieving null then, `Promise<R>` should be replaced by `Promise<Nullable<R>>` where `Nullable<T> = T | null`.

Reply:
`handleResponse` will be deleted in the [next PR](https://github.com/zuko4celik/movies-app-new/pull/45) caused by integrating Axios and interceptors. But, good comment to remember ;)